### DISCUSSION TITLE
Why is the Containerfile under tests?

### DISCUSSION THREAD

**Discussion by obnoxxx** [samba-in-kubernetes/sambacc#1] [General]
Hi @phlogistonjohn, when looking how the container is made that we use for building the samba-container inhttps://github.com/samba-in-kubernetes/samba-container/blob/master/images/server/Dockerfile.fedora, I noticed that this container definition is under `tests/container`. I was wondering whether this should be somewhere else than under tests.

**Comment by phlogistonjohn** [comment]
It's an outgrowth of how I started using the container: first as a way to run tests that have dependencies on the environment and other packages like samba's python modules. I then reused it to build the wheel to quickly get the sambacc code onto the samba-container image.
I have no issue with moving it, but its main purpose is still tests and I don't know how much of a long term thing it will be as part of the build process of the samba container. So I'm somewhat ambivalent here. :-)

**Reply by obnoxxx** [reply]
> It's an outgrowth of how I started using the container: first as a way to run tests that have dependencies on the environment and other packages like samba's python modules. I then reused it to build the wheel to quickly get the sambacc code onto the samba-container image.
> I have no issue with moving it, but its main purpose is still tests and I don't know how much of a long term thing it will be as part of the build process of the samba container.

Ok, are there ideas on how to build the samba container differently?

> So I'm somewhat ambivalent here. :-)

**Comment by phlogistonjohn** [comment]
On Friday, January 29, 2021 3:13:30 AM EST Michael Adam wrote:
> > It's an outgrowth of how I started using the container: first as a way to
> > run tests that have dependencies on the environment and other packages
> > like samba's python modules. I then reused it to build the wheel to
> > quickly get the sambacc code onto the samba-container image. I have no
> > issue with moving it, but its main purpose is still tests and I don't
> > know how much of a long term thing it will be as part of the build
> > process of the samba container.
> Ok, are there ideas on how to build the samba container differently?

In the future, we could build the wheel, or other package format (source 
tarball, rpm, etc), ahead of time as part of a release process and have it on 
github releases or pypi.

Then the samba container wouldn't need to use the sambacc container as a build 
container first. It could pull down the sambacc package directly.


There are probably some other things I have not thought of that may work too.
